### PR TITLE
This PR contains several features and fixes to react-kit and rx-utils

### PR DESCRIPTION
**Pull request by raveclassic**
see commit list

### CODE REVIEW

File: packages/rx-utils/src/live-data.utils.ts [hunk 1]
```diff
@@ -26,13 +30,50 @@ const tupleC = <A>(a: A) => <B>(b: B): [A, B] => [a, b];
...
 
-export const instanceLiveData: MonadThrow2<URI> & CoproductLeft<URI> = {
+export const instanceLiveData: MonadThrow2<URI> & CoproductLeft<URI> & Filterable2<URI> = {
 	URI,
 	...getApplicativeComposition(instanceObservable, remoteData),
 	chain: (fa, f) => instanceObservable.chain(fa, a => (isSuccess(a) ? f(a.value) : of(a))),
 	throwError: e => of(failure(e)),
 	coproductLeft: <EA, A, EB, B>(fa: LiveData<EA, A>, fb: LiveData<EB, B>) =>
 		instanceObservable.map(combineLatest(fa, fb), ([a, b]) => combineRD(a, b)),
+	filter: <E, A>(fa: LiveData<E, A>, p: Predicate<A>): LiveData<E, A> =>
```
Comment (scink) [inline packages/rx-utils/src/live-data.utils.ts]: 👍
was missing it so much

Comment (raveclassic) [reply packages/rx-utils/src/live-data.utils.ts]: > 👍
> was missing it so much

note that it's different from usual `filterRD` - it passes inner RemoteData as is if not RemoteSuccess

Comment (scink) [reply packages/rx-utils/src/live-data.utils.ts]: that is just what i need. i wrote a helpers yesterday with exact behaviour :)

**Comment by scink** [comment]
@raveclassic, what is `useSubscription` for? is it like useObservable but does not force component to rerender infinitely?

**Comment by raveclassic** [comment]
> @raveclassic, what is `useSubscription` for? is it like useObservable but does not force component to rerender infinitely?

It's for tracking subscription to arbitrary observable without manipulating the value. Just `subscribe`/`unsubscribe`. 
`useObservable` on the other hand keeps the value from observable in its inner state and exposes it as the hook's return value (also accepting initial value).

**Comment by raveclassic** [comment]
> is there any replacement of EntityStore?

Yeah, latest projects don't use EntityStore because of its overcomplicated logic and because it holds too much responsibility. In 99% cases you just need some aggregating structure (for example a `Map`) inside simple `Observable` or `LiveData`.

File: packages/react-kit/src/utils/use-observable.ts [hunk 2]
```diff
@@ -17,15 +17,18 @@ export function useObservable<M extends URIS2>(M: MonadObservable2<M>): <E, A>(f
...
-		const [state, setState] = useState(initial);
-		useEffect(() => {
-			const subscription = M.subscribe(fa, {
-				next: setState,
-				end: constVoid,
-			});
-			return () => subscription.unsubscribe();
-		}, [fa, setState]);
-		return state;
+		const [value, setValue] = useState(() => initial);
```
Comment (Nedgeva) [inline packages/react-kit/src/utils/use-observable.ts]: Thanks for fixing useObservable 👍

File: packages/rx-utils/src/live-data.utils.ts [hunk 3]
```diff
@@ -26,13 +30,50 @@ const tupleC = <A>(a: A) => <B>(b: B): [A, B] => [a, b];
...
+					const b = f(a.value);
+					if (isSome(b)) {
+						subscriber.next(success(b.value));
+					}
+				}
+			}),
+		),
+	compact: fa => instanceLiveData.filterMap(fa, identity),
+	partition: <E, A>(fa: LiveData<E, A>, p: Predicate<A>): Separated<LiveData<E, A>, LiveData<E, A>> => ({
+		left: instanceLiveData.filter(fa, not(p)),
```
[the remark this answers is not in the document]
Comment (raveclassic) [reply packages/rx-utils/src/live-data.utils.ts]: tuple? not sure I get you. `partition` requires to return `Separated` structure

**Pull request merged**